packages/thirdweb/src/react/web/ui/ConnectWallet/screens/SignatureScreen.tsx

```diff
@@ -0,0 +1,321 @@
+import { keyframes } from "@emotion/react";
+import { ReloadIcon } from "@radix-ui/react-icons";
+import { useCallback, useEffect, useRef, useState } from "react";
+import type { ThirdwebClient } from "../../../../../client/client.js";
+import type { Wallet } from "../../../../../wallets/interfaces/wallet.js";
+import { useSiweAuth } from "../../../../core/hooks/auth/useSiweAuth.js";
+import {
+  useActiveWallet,
+  useDisconnect,
+} from "../../../../core/hooks/wallets/wallet-hooks.js";
+import { wait } from "../../../../core/utils/wait.js";
+import { LoadingScreen } from "../../../wallets/shared/LoadingScreen.js";
+import { Spacer } from "../../components/Spacer.js";
+import { Spinner } from "../../components/Spinner.js";
+import { WalletImage } from "../../components/WalletImage.js";
+import { Container, ModalHeader } from "../../components/basic.js";
+import { Button } from "../../components/buttons.js";
+import { Text } from "../../components/text.js";
+import { useCustomTheme } from "../../design-system/CustomThemeProvider.js";
+import { StyledDiv } from "../../design-system/elements.js";
+import { iconSize, radius, spacing } from "../../design-system/index.js";
+import type { ConnectButtonProps } from "../ConnectWalletProps.js";
+import { TOS } from "../Modal/TOS.js";
+import type { ConnectLocale } from "../locale/types.js";
+import { WalletLogoSpinner } from "./WalletLogoSpinner.js";
+
+type Status = "signing" | "failed" | "idle";
+
+export const SignatureScreen: React.FC<{
+  onDone: () => void;
+  modalSize: "compact" | "wide";
+  termsOfServiceUrl?: string;
+  privacyPolicyUrl?: string;
+  connectLocale: ConnectLocale;
+  client: ThirdwebClient;
+  auth: ConnectButtonProps["auth"];
+}> = (props) => {
+  const {
+    onDone,
+    modalSize,
+    termsOfServiceUrl,
+    privacyPolicyUrl,
+    connectLocale,
+  } = props;
+
+  const siweAuth = useSiweAuth(props.auth);
+  const [status, setStatus] = useState<Status>("idle");
+  const { disconnect } = useDisconnect();
+  const wallet = useActiveWallet();
+  const locale = connectLocale.signatureScreen;
+
+  const signIn = useCallback(async () => {
+    try {
+      setStatus("signing");
+      await wait(1000);
```

Comment:
why the artificial delay?

Reply:
so that It doesn't "flash" the screen if it fails immediately - With this, we always show a Loading spinner for at least 1s to "ease" into failure